### PR TITLE
fix(router): add set-misc module to nginx build

### LINKED ISSUE #3964: AffinityArg hangs router on 1.7.3
I'm trying to apply the affinityArg setting on the router…

```
deisctl config router set affinityArg=sid
```

…but I keep getting this error on the logs:

```
[INFO] - 2015/07/04 09:01:12 [emerg] 136#0: unknown directive "set_random" in /opt/nginx/conf/.nginx.conf102133528:140
[INFO] - waiting for confd to write initial templates...
```

If I remove the setting, everything goes back as normal.

Have I done something wrong?

### PR DESCRIPTION
**Pull request by mboersma**
The commit c7c895c fixes the affinityArg hashing issue by using the `set_random` directive on the nginx configuration, but forgets to include the module in which it is declared. This commit fixes this, while also tries to make the build script a little cleaner.

Closes #3964.

Replaces #3965--thanks @fabiob!

### CODE REVIEW

**Comment by carmstrong** [comment]
Code LGTM.

File: router/build.sh [hunk 1]
```diff
@@ -38,14 +40,11 @@ apk add --update-cache \
...
-curl -sSL https://github.com/nbs-system/naxsi/archive/$VERSION_NAXSI.tar.gz -o $BUILD_PATH/$VERSION_NAXSI.tar.gz
-
-# expand the source files
-tar xzf $VERSION_NGINX.tar.gz
-tar xzf $VERSION_NAXSI.tar.gz
+# download and extract the source files
+curl -sSL http://nginx.org/download/$VERSION_NGINX.tar.gz | tar xz
+curl -sSL https://github.com/nbs-system/naxsi/archive/$VERSION_NAXSI.tar.gz | tar xz
+curl -sSL https://github.com/simpl/ngx_devel_kit/archive/v$VERSION_NDK.tar.gz | tar xz
+curl -sSL https://github.com/openresty/set-misc-nginx-module/archive/v$VERSION_SETMISC.tar.gz | tar xz
```
Comment (sgoings) [inline router/build.sh]: Why not loop around your sources, then `curl <item> | tar xz`?

Comment (mboersma) [inline router/build.sh]: That could be more DRY, but I'm ok with this minor instance of repetition unless @fabiob is motivated to rewrite it.

Comment (fabiob) [inline router/build.sh]: I plan to rewrite this to include @bladealslayer changes on #3954 – also, to include verification of the plugins as well

Comment (fabiob) [inline router/build.sh]: Actually, I just noticed that the current rebase removes @bladealslayer's changes, so the rewrite is a must.

Comment (mboersma) [inline router/build.sh]: Good catch. Restoring that and adding verification of the other tarballs would be great. :green_heart:

Comment (fabiob) [inline router/build.sh]: See the new PR: #4028

**Comment by mboersma** [comment]
Closing in favor of #4028.

**Pull request closed without merging**